# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1295043683 28800
#      Fri Jan 14 14:21:23 2011 -0800
# Node ID 06c80c90a2466ba904ba9aeef113296d6726662b
# Parent  34bb8d38f19f01ab446f90e76e4f0bc57762aeb5
Add undocumented function svd_driver to documentation.

diff --git a/doc/ChangeLog b/doc/ChangeLog
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -1,8 +1,12 @@
+2011-01-14  Rik  <octave@nomad.inbox5.com>
+
+	* interpreter/linalg.txi: Add undocumented svd_driver to documentation.
+
 2011-01-14  Rik  <octave@nomad.inbox5.com>
 
 	* interpreter/arith.txi: Add undocumented cbrt to documentation.
 
 2011-01-14  Rik  <octave@nomad.inbox5.com>
 
 	* interpreter/arith.txi: Add undocumented erfcx to documentation.
 
diff --git a/doc/interpreter/linalg.txi b/doc/interpreter/linalg.txi
--- a/doc/interpreter/linalg.txi
+++ b/doc/interpreter/linalg.txi
@@ -154,16 +154,18 @@ flag a non-Hermitian matrix.
 @DOCSTRING(qzhess)
 
 @DOCSTRING(schur)
 
 @DOCSTRING(subspace)
 
 @DOCSTRING(svd)
 
+@DOCSTRING(svd_driver)
+
 @c FIXME -- should there be a new section here?
 
 @DOCSTRING(housh)
 
 @DOCSTRING(krylov)
 
 @node Functions of a Matrix
 @section Functions of a Matrix
diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2011-01-14  Rik  <octave@nomad.inbox5.com>
+
+	* DLD-FUNCTIONS/svd.cc: Add Seealso references to svd.
+
 2011-01-14  Rik  <octave@nomad.inbox5.com>
 
 	* src/mappers.cc: Add Seealso links between sqrt, cbrt, nthroot
 	functions.
 
 2011-01-14  John W. Eaton  <jwe@octave.org>
 
 	* dirfns.cc (Ffnmatch): Use DEFUNX until gnulib's fnmatch is C++
diff --git a/src/DLD-FUNCTIONS/svd.cc b/src/DLD-FUNCTIONS/svd.cc
--- a/src/DLD-FUNCTIONS/svd.cc
+++ b/src/DLD-FUNCTIONS/svd.cc
@@ -116,16 +116,17 @@ v =\n\
   -0.45986  -0.52829  -0.71375\n\
   -0.32330  -0.64901   0.68867\n\
 @end group\n\
 @end example\n\
 \n\
 If given a second argument, @code{svd} returns an economy-sized\n\
 decomposition, eliminating the unnecessary rows or columns of @var{u} or\n\
 @var{v}.\n\
+@seealso{svd_driver, svds, eig}\n\
 @end deftypefn")
 {
   octave_value_list retval;
 
   int nargin = args.length ();
 
   if (nargin < 1 || nargin > 2 || nargout == 2 || nargout > 3)
     {
@@ -401,17 +402,18 @@ decomposition, eliminating the unnecessa
 %!error <Invalid call to svd.*> svd ([1, 2; 4, 5], 2, 3);
 %!error <Invalid call to svd.*> [u, v] = svd ([1, 2; 3, 4]);
 
 */
 
 DEFUN_DLD (svd_driver, args, nargout,
   "-*- texinfo -*-\n\
 @deftypefn {Loadable Function} {@var{old} =} svd_driver (@var{new})\n\
-Sets or queries the underlying @sc{lapack} driver used by svd.\n\
-Currently recognized are \"gesvd\" and \"gesdd\".  Default is \"gesvd\".\n\
+Set or query the underlying @sc{lapack} driver used by @code{svd}.\n\
+Currently recognized values are \"gesvd\" and \"gesdd\".  The default\n\
+is \"gesvd\".\n\
 @seealso{svd}\n\
 @end deftypefn")
 {
   static const char *driver_names[] = { "gesvd", "gesdd", 0 };
 
   return SET_INTERNAL_VARIABLE_CHOICES (svd_driver, driver_names);
 }
